src/convert.jl

```diff
@@ -103,6 +112,14 @@ function trixi2vtk(filename::AbstractString...;
     verbose && println("| Reading mesh file...")
     @timeit "read mesh" mesh = Trixi.load_mesh_serial(meshfile; n_cells_max=0, RealT=Float64)
 
+    # Check compatibility of the mesh type and the output format
+    if format == :vti && typeof(mesh) != Trixi.TreeMesh{2, Trixi.SerialTree{2}}
+      error("VTI format only available for 2D TreeMesh")
+    end
+
+    # Create an empty `node_set` such that converting a mesh.h5 file also works
+    node_set = []
```

Comment:
This is definitely not type stable (since it creates a `Vector{Any}` and is later overwritten). Not sure if it's an issue, but it would be better to put this in the `else` clause of `if is_datafile`

Reply:
I am also not sure if the type instability is an issue. I need to remember exactly what the issue was that this fixed regarding the conversion. I can also try putting it into an `is_datafile` statement.

Reply:
This appears to be leftover from an earlier version. I just removed it and all the tests pass locally. I will commit and push this change to make sure the tests pass on github.